### ISSUE TITLE
CKV2_AZURE_41 false positive?

### ISSUE THREAD

**Issue by acelebanski** [bridgecrewio/checkov#6140]
**Describe the issue**
I applied the fix to `CKV2_AZURE_41` check according to [this](https://docs.prismacloud.io/en/enterprise-edition/policy-reference/azure-policies/azure-iam-policies/bc-azure-2-41) guide. Unfortunately, the check still fails.

**Examples**
In code it looks like that:

```hcl
resource "azurerm_storage_account" "this" {
  name = var.name
  (...)

  sas_policy {
    expiration_period = "90.00:00:00"
    expiration_action = "Log"
  }
}
```

**Version (please complete the following information):**
 - Checkov Version **3.2.50**

**Comment by naveednawazkhan** [comment]
Hi @acelebanski, I have tested this policy and do not see any issue with code but I have requested our team to update the [docs](https://github.com/hlxsites/prisma-cloud-docs/pull/568/commits/104fa9bacdd0025cbe6a5524d6a092713f015c13). 

I used the following code for testing.


`resource "azurerm_storage_account" "pass_1" {
  name                     = "pud-storage2023abc1"
  resource_group_name      = var.rg-name
  location                 = var.location
  account_tier             = "Standard"
  account_replication_type = "GRS"
  shared_access_key_enabled = true

  sas_policy {
   expiration_period = "90.00:00:00"
    expiration_action = "Log"
  }

  tags = {
    bc_status = "pass"
  }
}`


If the issue persists, please share the full resource code. Thanks

**Comment by alexivanov-danone** [comment]
Same issue

**Comment by acelebanski** [comment]
It looks like it's not fixed yet, I tried to add this `sas_policy` block to this repo:
https://github.com/PaloAltoNetworks/terraform-azurerm-swfw-modules/blob/main/modules/bootstrap/main.tf

But the check still doesn't pass so I had to ignore it.

**Comment by acelebanski** [comment]
Hello @naveednawazkhan, any news on this one?

**Comment by bostrowski13** [comment]
still an issue, and doesnt seem to like the use of variables.

```
  sas_policy {
    expiration_period = var.storage_account_sas_expiration_period
    expiration_action = "Log"
  }
```

**Comment by naveednawazkhan** [comment]
@acelebanski, Really sorry for the delay, can you try adding `shared_access_key_enabled = true`? If `shared_access_key_enabled = false` then no need to set `sas_policy` but if `shared_access_key_enabled = true` then `sas_policy `can be used to avoid fail.

Example:
```
sas_policy {
   expiration_period = "90.00:00:00"
    expiration_action = "Log"
  }
```
By reviewing policy code, it seems that `shared_access_key_enabled` is mandatory. It either has to be false/true.

If `shared_access_key_enabled` is missing, regardless we add sas_policy it'll fail.

**Comment by sluscombepure** [comment]
@naveednawazkhan Thanks. Explicitly adding `shared_access_key_enabled = true` fixed the error for me.

**Comment by naveednawazkhan** [comment]
Awesome! Also, I created a PR to improve policy descriptions to avoid confusion in the future.

**Comment by daviewales** [comment]
@naveednawazkhan, I think it's a bug that `shared_access_key_enabled = true` must be specified in the TF configuration for this to pass, as [shared_access_key_enabled](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account#shared_access_key_enabled-1) is an optional value which defaults to `true`. So if the key is missing, then it should be exactly the same as specifying the key with `shared_access_key_enabled = true`.
